app/data/domain/dog.js

```diff
@@ -56,6 +77,26 @@ class Dog extends Changeable {
     this._updates.update('status', status, callback)
     this._status = status
   }
+
+  /**
+   * @return {BreachCategory[]}
+   */
+  get breaches () {
+    return this._breaches
+  }
+
+  setBreaches (breaches, allDogBreaches, callback) {
+    const dogBreachDictionary = allDogBreaches.reduce((dogBreachDict, breach) => {
+      return {
+        ...dogBreachDict,
+        [breach.short_name]: breach
+      }
+    }, {})
+    const dogBreachBreaches = breaches.map(breach => dogBreachDictionary[breach])
+    this._breaches = dogBreachBreaches
+    this._updates.update('dogBreaches', dogBreachBreaches, undefined)
+    this.setStatus('In breach', callback)
```

Comment:
We now have a /app/constants/statuses.js (from a previous PR) so you can reference the status 'In breach' as a constant